camel/models/cohere_model.py

```diff
@@ -0,0 +1,240 @@
+            model=self.model_type.value,
+            object="chat.completion",
+            usage={
+                "prompt_tokens": safe_get(
+                    response, 'meta', 'billed_tokens', 'prompt_tokens'
+                )
+                or 0,
+                "completion_tokens": safe_get(
+                    response, 'meta', 'billed_tokens', 'completion_tokens'
+                )
+                or 0,
+                "total_tokens": (
+                    (
+                        safe_get(
+                            response, 'meta', 'billed_tokens', 'prompt_tokens'
+                        )
+                        or 0
+                    )
+                    + (
+                        safe_get(
+                            response,
+                            'meta',
+                            'billed_tokens',
+                            'completion_tokens',
+                        )
+                        or 0
+                    )
+                ),
+            },
+        )
+        return obj
+
+    def _to_cohere_chatmessage(
+        self, messages: List[OpenAIMessage]
+    ) -> List[Dict[str, str]]:
+        new_messages = []
+        for msg in messages:
+            role = msg.get("role")
+            content = msg.get("content")
+
+            if role == "user":
+                new_messages.append({"role": "User", "message": content})
+            elif role == "assistant":
+                new_messages.append({"role": "Chatbot", "message": content})
+            elif role == "system":
+                new_messages.append({"role": "System", "message": content})
+            else:
+                raise ValueError(f"Unsupported message role: {role}")
+
+        return new_messages  # type: ignore[return-value]
+
+    @property
+    def token_counter(self) -> BaseTokenCounter:
+        """Initialize the token counter for the model backend.
+
+        Returns:
+            BaseTokenCounter: The token counter following the model's
+                tokenization style.
+        """
+        if not self._token_counter:
+            self._token_counter = OpenAITokenCounter(
+                model=ModelType.GPT_3_5_TURBO
+            )
+        return self._token_counter
+
+    @api_keys_required("COHERE_API_KEY")
+    def run(self, messages: List[OpenAIMessage]) -> ChatCompletion:
+        """Runs inference of Cohere chat completion.
+
+        Args:
+            messages (List[OpenAIMessage]): Message list with the chat history
+                in OpenAI API format.
+
+        Returns:
+            ChatCompletion.
+        """
+        from cohere.core.api_error import ApiError
+
+        cohere_messages = self._to_cohere_chatmessage(messages)
+
+        # Filter out unsupported parameters
+        supported_params = {
+            'temperature',
+            'p',
+            'k',
+            'max_tokens',
+            'prompt_truncation',
+        }
```

Comment:
this part of check is done by `self.check_model_config()` in `BaseModelBackend`

Reply:
Got it. detele in new commit